SmartDeviceLink/SDLLockScreenManager.m

```diff
@@ -121,6 +136,14 @@ - (void)sdl_appDidBecomeActive:(NSNotification *)notification {
     [self sdl_checkLockScreen];
 }
 
+- (void)sdl_driverDistractionStateDidChange:(SDLRPCNotificationNotification *)notification {
```

Comment:
Please remove the `SDLLockScreenStatusManager` from `SDLProxy` and move the related code here instead.

Reply:
There is too much binding logic between `SDLLockScreenStatusManager` and `SDLProxy` because both `handleAfterHMIStatus` and `handleAfterDriverDistraction` call `invokeMethodOnDelegates` which is tied to the proxy object.
`SDLLockScreenManager` can observe for `SDLDidChangeHMIStatusNotification` and `SDLDidChangeDriverDistractionStateNotification` but it can't call `invokeMethodOnDelegates` cleanly
 I think we should leave this for now.

Reply:
No, this is all unnecessary. The `SDLLockScreenStatusManager` was designed before the `SDLManager` system existed and is now entirely extraneous. `handleAfterDriverDistraction` can be deleted, `handleAfterHMIStatus` can be deleted, and the `SDLLockScreenStatusManager` can be deleted. It's creating a "fake" RPC notification that doesn't exist (and can now be deleted as well). It's logic can all be moved into the `SDLLockScreenManager` by directly observing `onDriverDistraction` and `onHMIStatus` which it already does.